configs/dlstreamer/framework-pipelines/stream_density.sh

```diff
@@ -5,17 +5,80 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
+# this function cleans up parent process and its child processes
+# the first input is the parent process to be cleaned up
+cleanupPipelineProcesses()
+{
+	pidToKill=$1
+	childPids=$(pgrep -P $pidToKill)
+	echo "decrementing pipelines and to kill pid $pidToKill"
+	waitForChildPidKilled=0
+	if [ -z "$childPids" ]
+	then
+		echo "for parent pid $pidToKill, there is no child pids to kill"
+	else
+		echo "parent pid $pidToKill with childPids $childPids to be killed"
+		waitForChildPidKilled=1
+	fi
+
+	pkill -P $pidToKill
```

Comment:
Recommend adding a short comment like "# Kill parent process" and another at line 48 "# make sure parent pid is gone"